pkg/cloudprovider/azure.go

```diff
@@ -160,11 +162,24 @@ func (a *Azure) ReleasePrivateIP(ip net.IP, node *corev1.Node) error {
 	}
 	networkInterface.IPConfigurations = &keepIPConfiguration
 	// Send the request
-	result, err := a.createOrUpdate(networkInterface)
+	_, err = a.createOrUpdate(networkInterface)
 	if err != nil {
 		return err
 	}
-	return a.waitForCompletion(result)
+	return wait.PollImmediate(2*time.Second, time.Minute, func() (done bool, err error) {
+		networkInterfaces, err := a.getNetworkInterfaces(instance)
+		if err != nil {
+			return false, err
+		}
+		networkInterface := networkInterfaces[0]
+		assignedIPs := []string{}
```

Comment:
how about we just return for a retry instead of assembling a sets.NewString?
Like:
```
	return wait.PollImmediate(2*time.Second, 2 * time.Minute, func() (done bool, err error) {
		networkInterfaces, err := a.getNetworkInterfaces(instance)
		if err != nil {
			return false, err
		}
		networkInterface := networkInterfaces[0]
		for _, ipConfiguration := range *networkInterface.IPConfigurations {
			if assignedIP := net.ParseIP(*ipConfiguration.PrivateIPAddress); assignedIP != nil && assignedIP.Equal(ip) {
				return false, nil
			}
		}
		return true, nil
	})
```

Reply:
yes @flavio-fernandes , this is a fast performing loop, changed it.